### PR TITLE
Frontend Code Challenge

### PR DESCRIPTION
**Pull request by sanjaysingh1982**
- Task 1

1. Add test for patient search component.
2. Add loading spinner.

- Task 2

1. Error handling and improve the toast component.

### CODE REVIEW

File: src/components/patients/PatientsApi.ts [hunk 1]
```diff
@@ -1,12 +1,10 @@
...
-  PatientSearchQuery,
-} from "./patients";
+import { PatientService } from './../../models/PatientServiceModel';
+import { sleep } from "../../utils/sleep";
+import { Patient } from '../../models/PatientModel';
+import { isPatients } from './Patients';
+import { PatientSearchQuery } from '../../models/PatientSearchQueryModel';
 
-export function createNewPatientsApi(baseUrl: string): PatientsService {
+export function createNewPatientsApi(baseUrl: string): PatientService {
```
Comment (heartbeat-med) [inline src/components/patients/PatientsApi.ts:7]: Why was this renamed from `PatientsService` to `PatientService`?

File: src/components/patients/PatientsLoader.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import { FunctionComponent } from "react";
+import { Button } from "react-bootstrap";
+import { Patient } from "../../models/PatientModel";
+import { IToast } from "../../models/ToastModel";
+
+type props = {
+  loadPatients: () => Promise<Patient[]>;
+  onLoaded: (ps: Patient[]) => void;
+  setLoading: (loader: boolean) => void;
```
Comment (heartbeat-med) [inline src/components/patients/PatientsLoader.tsx:9]: Why is the loading state lifted up to the parent component, but the search request is made in this component? Is there an advantage to doing it this way?

File: src/components/spinner/SearchSpinner.tsx [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import { FunctionComponent } from "react";
+import { Spinner } from "react-bootstrap";
+
+export const SearchSpinner: FunctionComponent = () => {
```
Comment (heartbeat-med) [inline src/components/spinner/SearchSpinner.tsx:4]: What's the reasoning behind wrapping this component, what's the advantage?

File: src/models/PatientSearchQueryModel.tsx [hunk 4]
```diff
@@ -0,0 +1,7 @@
+// PatientSearchQuery, a search query which allows
```
Comment (heartbeat-med) [inline src/models/PatientSearchQueryModel.tsx]: Is there a reason why this file is `.tsx`, would `.ts`?

File: src/models/ToastModel.tsx [hunk 5]
```diff
@@ -0,0 +1,5 @@
+export interface IToast {
+    title?: string;
+    message?: string;
+    status?: "success" | "danger";
```
Comment (heartbeat-med) [inline src/models/ToastModel.tsx]: Is there a reason to make `status` and `title` optional?

**Review by heartbeat-med — COMMENTED** [review]
Great work, we have left some comments in this review. Please either submit more code to address then, reply or come prepared to the next interview and we will go through these together. We will contact you via email to schedule the next call!

File: src/components/patients/PatientsApi.ts (hunk 1 above)
Comment (sanjaysingh1982) [reply src/components/patients/PatientsApi.ts:7]: Just to keep the name singular and it's an interface.

File: src/components/patients/PatientsLoader.tsx (hunk 2 above)
Comment (sanjaysingh1982) [reply src/components/patients/PatientsLoader.tsx:9]: To show the loading spinner on the click of a button "Load all patients" and also on the search text input.

File: src/components/spinner/SearchSpinner.tsx (hunk 3 above)
Comment (sanjaysingh1982) [reply src/components/spinner/SearchSpinner.tsx:4]: The actual advantage we will get when we scale this product. We can use this at different screens and can be more customized based on the different types of spinners like progress bar etc.

File: src/models/PatientSearchQueryModel.tsx (hunk 4 above)
Comment (sanjaysingh1982) [reply src/models/PatientSearchQueryModel.tsx]: No reason. Just missed changing it 😄 
09b89eb43b17830307b924df2a92351a73036380

File: src/models/ToastModel.tsx (hunk 5 above)
Comment (sanjaysingh1982) [reply src/models/ToastModel.tsx]: Yes, because I don't want to feed the ToastBox component with the default values.
